packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx

```diff
@@ -0,0 +1,393 @@
+});
+
+test('Shows the overflowing tab count when showTabCount is true', () => {
+  render(
+    <OverflowTab
+      overflowingTabs={[
+        { title: 'Tab one', eventKey: 1 },
+        { title: 'Tab two', eventKey: 2 }
+      ]}
+      showTabCount
+    />
+  );
+
+  expect(screen.getByRole('tab', { name: 'More (2)' })).toBeVisible();
+});
+
+test('Uses "More" as the default overflow tab title', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByRole('tab')).toHaveTextContent('More');
+});
+
+test('Uses the selected tab title as the overflow tab title rather than the default title when a tab is selected', () => {
+  render(
+    <TabsContext.Provider value={{ ...tabsContextDefaultProps, localActiveKey: 1 }}>
+      <OverflowTab
+        overflowingTabs={[
+          { title: 'Tab one', eventKey: 1, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> }
+        ]}
+      />
+    </TabsContext.Provider>
+  );
+
+  expect(screen.getByRole('tab')).toHaveTextContent('Tab one');
+});
+
+test('Uses custom overflow tab title text provided via defaultTitleText', () => {
+  render(<OverflowTab defaultTitleText="Test" />);
+
+  expect(screen.getByRole('tab')).toHaveTextContent('Test');
+});
+
+test('Uses the selected tab title as the overflow tab title rather than the defaultTitleText when a tab is selected', () => {
+  render(
+    <TabsContext.Provider value={{ ...tabsContextDefaultProps, localActiveKey: 1 }}>
+      <OverflowTab
+        overflowingTabs={[
+          { title: 'Tab one', eventKey: 1, tabContentRef: ('fakeRef' as unknown) as React.RefObject<any> }
+        ]}
+        defaultTitleText="Test"
+      />
+    </TabsContext.Provider>
+  );
+
+  expect(screen.getByRole('tab')).toHaveTextContent('Tab one');
+});
+
+test('Does not have an aria label on the tab by default', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByRole('tab')).not.toHaveAttribute('aria-label');
+});
+
+test('Renders with the toggleAriaLabel as an aria label on the tab', () => {
+  render(<OverflowTab toggleAriaLabel="Test aria label" />);
+
+  expect(screen.getByRole('tab')).toHaveAttribute('aria-label', 'Test aria label');
+});
+
+test('Renders the tab with aria-haspopup=menu', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByRole('tab')).toHaveAttribute('aria-haspopup', 'menu');
+});
+
+test('Renders the tab with aria-expanded set to false by default', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByRole('tab')).toHaveAttribute('aria-expanded', 'false');
+});
+
+test('Renders the tab with aria-expanded set to true when the menu is opened', () => {
+  render(<OverflowTab />);
+
+  const overflowTab = screen.getByRole('tab');
+
+  userEvent.click(overflowTab);
+
+  expect(overflowTab).toHaveAttribute('aria-expanded', 'true');
+});
+
+test('Passes Popper popperMatchesTriggerWidth set to false', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByText('Popper matches trigger width: false')).toBeVisible();
```

Comment:
nit: I might reference the mocked Popper above here in some way maybe via comment since it's so far away from that mock and it might not be obvious jumping to this test where the "Popper matches trigger width" text comes from. Same with the tests below.